### PR TITLE
Fix #5: maven property are not resolved against property of directory-of goal

### LINKED ISSUE #5: maven property are not resolved against property of directory-of goal
Hi, I'm using your plugin to define a directory of a module where are placed some integration tests executed by antrun plugin.

I run in the same reactor also the sonarqube analisys software.
I need to specify a sonar property where is located the the folder where the JUnit tests are placed. If I use a property in the parent pom as like

    <sonar.reportPath>distribution/server.test/target/surefire-tests</sonar.reportPath>

each module relocate to its baseDir so it does not work.

So I add your directory-of goal in the parent pom. The above property became

    <sonar.reportPath>${test.project}/target/surefire-tests</sonar.reportPath>

I thought it worked but the property is not resolved against the new property created with your plugin.

### PR DESCRIPTION
**Pull request by nfalco79**
After add the property to the current project, all the current project
properties are resolved against the new one.

### CODE REVIEW

**Comment by nfalco79** [comment]
Hi, man.
Could I suggest that your Mojo class extends AbstractMojo class instead implement the interfaces Mojo, ContextEnabled to avoid boiler template code already done in the Abstract class?

**Comment by mgh87** [comment]
Hey.
I just ran into the same problem and was wondering if you plan to merge this pull request that should solve it.
Best regards

File: src/main/java/org/commonjava/maven/plugins/execroot/AbstractDirectoryGoal.java [hunk 1]
```diff
@@ -165,4 +169,28 @@ public void setPluginContext( final Map context )
...
+
+		for (Enumeration<?> n = projectProperties.propertyNames(); n.hasMoreElements();) {
+			String k = (String) n.nextElement();
+			if (!k.equals(property)) {
+				projectProperties.setProperty(k, getPropertyValue(k, projectProperties));
+			}
+		}
+	}
+
+	private String getPropertyValue(String key, Properties properties) {
```
Comment (jdcasey) [inline src/main/java/org/commonjava/maven/plugins/execroot/AbstractDirectoryGoal.java]: This really needs to use StringSearchInterpolator, so we can handle expressions whose resolved values contain expressions. I'll try to take a look.

**Comment by jdcasey** [comment]
@nfalco79 Generally I try to use the interface to avoid problems with IncompatibleClassChangeError or similar if the plugin gets used with a version of Maven that was compiled in an incompatble way.

**Comment by jdcasey** [comment]
Okay, I'm looking through this code, and I'm not clear on what's happening that's incorrect. The original description is pretty light. It looks like you want this newly-defined property to be usable for re-evaluating project properties. Is that correct?

Can someone add a failing test case (even just a zip containing pom's, etc.) to show why this is broken, and what the expected result is? It should be possible to attach something like this using:

https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/

**Comment by nfalco79** [comment]
Closed as we moved time ago to ${session.executionRootDirectory}

**Pull request closed without merging**